public/app/features/alerting/partials/alert_tab.html

```diff
@@ -20,6 +20,7 @@ <h4 class="section-heading">Rule</h4>
           type="text"
           ng-model="ctrl.alert.frequency"
           ng-blur="ctrl.checkFrequency()"
+          ng-pattern="/(^\d+([dhms])$)|(0)|(^$)/"
```

Comment:
Frequency shouldn't allow an empty or zero input. Sorry if I wasn't clear on that.